### PR TITLE
Remove dependency on httpbin.org and run local responses server

### LINKED ISSUE #3101: xhr native and polyfill flake
```
Chromium 45.0.2454 (Ubuntu 0.0.0) XHR Native should do simple JSON fetch FAILED
	Error: timeout of 5000ms exceeded. Ensure the done() callback is being called in this test.
Chromium 45.0.2454 (Ubuntu 0.0.0): Executed 1377 of 1991 (1 FAILED) (0 secs / 13.411 secs)
```

```
Chromium 45.0.2454 (Ubuntu 0.0.0) XHR Polyfill should do simple JSON fetch FAILED
	Error: timeout of 5000ms exceeded. Ensure the done() callback is being called in this test.
Chromium 45.0.2454 (Ubuntu 0.0.0): Executed 1400 of 1991 (2 FAILED) (0 secs / 18.802 secs)
```

https://travis-ci.org/ampproject/amphtml/builds/127920478

### PR DESCRIPTION
**Pull request by mkhatib**
Fixes #3101 and #4899

### CODE REVIEW

**Comment by mkhatib** [comment]
yay! no flakes!

File: test/functional/test-xhr.js [hunk 1]
```diff
@@ -22,6 +22,8 @@ import {
   FetchResponse,
   assertSuccess,
 } from '../../src/service/xhr-impl';
+import {getCookie} from '../../src/cookies';
+
 
 describe('XHR', function() {
```
Comment (cramforce) [inline test/functional/test-xhr.js]: You need to rebase. I turned off these tests.

File: build-system/tasks/runtime-test.js [hunk 2]
```diff
@@ -22,6 +22,14 @@ var karmaConfig = config.karma;
 var extend = require('util')._extend;
 var fs = require('fs');
 var path = require('path');
+var util = require('gulp-util');
+var webserver = require('gulp-webserver');
+var app = require('../test-server').app;
+
+var host = argv.testResponsesServerHost || 'localhost';
+var port = argv.testResponsesServerPort || 9987;
```
Comment (cramforce) [inline build-system/tasks/runtime-test.js]: You don't really allow overriding this in the test. Lets use something higher and more random like `31862` :)

File: build-system/tasks/runtime-test.js [hunk 3]
```diff
@@ -22,6 +22,14 @@ var karmaConfig = config.karma;
...
+var useHttps = argv.https != undefined;
```
Comment (cramforce) [inline build-system/tasks/runtime-test.js]: Do we need this overridable? Otherwise I'd prefer to just kill these flags.

**Comment by cramforce** [comment]
This is great!!! A few little comments (and the rebase and subsequent removal of the `skip` is important), but lets get this in quickly!!!

File: build-system/tasks/runtime-test.js (hunk 3 above)
Comment (mkhatib) [inline build-system/tasks/runtime-test.js]: Done

File: build-system/tasks/runtime-test.js (hunk 2 above)
Comment (mkhatib) [inline build-system/tasks/runtime-test.js]: Done

File: test/functional/test-xhr.js (hunk 1 above)
Comment (mkhatib) [inline test/functional/test-xhr.js]: Done

**Comment by mkhatib** [comment]
Done PTAL 👀

**Comment by cramforce** [comment]
LGTM

**Comment by cramforce** [comment]
Merged it for you, so we get the tests back on ASAP.

**Comment by lannka** [comment]
@mkhatib have you considered using [Sinon FakeServer](http://sinonjs.org/docs/#fakeServer)? I believe launching this additional test server does add some latency. 

If you really need a real HTTP server for test, I think [Karma custom server middleware](http://karma-runner.github.io/1.0/config/configuration-file.html) is a better option.

**Pull request merged**